### PR TITLE
Drop --enable-display-list flag, remove associated branched code

### PR DESCRIPTION
**Pull request by dnfield**
Drops --enable-display-list flag and related code branches

~~Keeps PictureLayer around for now until we can figure out what to do with the SKP deserialization test harness. @akbiggs may know if that is safe to delete - but I'm fine with doing that in a follow up PR since this one is likely enough as it is to break things.~~ Deletes PictureLayer and a test harness that deserializes SKPs and uses PictureLayers as well.

Fixes https://github.com/flutter/flutter/issues/86634

Currently includes https://github.com/flutter/engine/pull/34227

Fixes https://github.com/flutter/flutter/issues/106435

### CODE REVIEW

**Review by chinmaygarde — APPROVED** [review]
Yes! This has caused a fair amount of confusion for me.

**Comment by flar** [comment]
Is the description accurate? It says it is keeping PictureLayer, but the file diffs show it deleted.

File: shell/common/shell_unittests.cc [hunk 1]
```diff
@@ -2386,33 +2376,32 @@ TEST_F(ShellTest, RasterizerMakeRasterSnapshot) {
...
 
-static sk_sp<SkPicture> MakeSizedPicture(int width, int height) {
-  SkPictureRecorder recorder;
-  SkCanvas* recording_canvas =
-      recorder.beginRecording(SkRect::MakeXYWH(0, 0, width, height));
-  recording_canvas->drawRect(SkRect::MakeXYWH(0, 0, width, height),
-                             SkPaint(SkColor4f::FromColor(SK_ColorRED)));
-  return recorder.finishRecordingAsPicture();
+static sk_sp<DisplayList> MakeSizedDisplayList(int width, int height) {
+  DisplayListCanvasRecorder recorder(SkRect::MakeXYWH(0, 0, 80, 80));
```
Comment (flar) [inline shell/common/shell_unittests.cc:2380]: width,height, not 80,80

Comment (flar) [reply shell/common/shell_unittests.cc:2380]: Why isn't this used in all of the above tests?

**Review by flar — CHANGES_REQUESTED** [review]

**Comment by flar** [comment]
Mostly only 1 "oops" and a couple of opportunities for deepening the cut, but feel free to file follow-on issues for the other stuff. Can we get away with no skp warmup tesst? Are those obsolete?

File: shell/common/shell_unittests.cc [hunk 2]
```diff
@@ -744,18 +744,17 @@ TEST_F(ShellTest, ExternalEmbedderNoThreadMerger) {
   RunEngine(shell.get(), std::move(configuration));
 
   LayerTreeBuilder builder = [&](std::shared_ptr<ContainerLayer> root) {
-    SkPictureRecorder recorder;
-    SkCanvas* recording_canvas =
-        recorder.beginRecording(SkRect::MakeXYWH(0, 0, 80, 80));
-    recording_canvas->drawRect(SkRect::MakeXYWH(0, 0, 80, 80),
-                               SkPaint(SkColor4f::FromColor(SK_ColorRED)));
-    auto sk_picture = recorder.finishRecordingAsPicture();
+    DisplayListCanvasRecorder recorder(SkRect::MakeXYWH(0, 0, 80, 80));
```
[the remark this answers is not in the document]
Comment (dnfield) [reply shell/common/shell_unittests.cc:747]: https://github.com/flutter/flutter/issues/106448

File: shell/common/shell_unittests.cc (hunk 1 above)
Comment (dnfield) [reply shell/common/shell_unittests.cc:2380]: Done, and updated other tests to use this.

File: lib/ui/painting/canvas.cc [hunk 3]
```diff
@@ -589,13 +589,7 @@ void Canvas::drawPicture(Picture* picture) {
...
     return;
   }
-  if (picture->picture()) {
-    if (display_list_recorder_) {
-      builder()->drawPicture(picture->picture(), nullptr, false);
-    } else if (canvas_) {
-      canvas_->drawPicture(picture->picture().get());
-    }
-  } else if (picture->display_list()) {
+  if (picture->display_list()) {
```
[the remark this answers is not in the document]
Comment (dnfield) [reply lib/ui/painting/canvas.cc:592]: Yes. I'll update canvas.cc

[the remark this answers is not in the document]
Comment (dnfield) [reply lib/ui/painting/canvas.cc:592]: it will be null after disposal though, so we still need the if guards to avoid segfaulting.

**Comment by dnfield** [comment]
The SKP warmup tests were added for Fuchsia by an engineer who's no longer working on Fuchsia. The engineers who are working on Fuchsia are telling me they don't think we need them.

**Review by flar — APPROVED** [review]
LGTM!

**Comment by zanderso** [comment]
Nice!

**Pull request merged**